### PR TITLE
Add last checked date

### LINKED ISSUE #50: Add last checked date field to item source

### LINKED ISSUE #51: Show last checked date for each source
Depends on #50.  
This is required to comply with http://www.barnivore.com/terms#reuse.

### PR DESCRIPTION
**Pull request by bash**
## Issues
- [x] Resolves #50
- [x] Resolves #51

### CODE REVIEW

File: src/item_loader.rs [hunk 1]
```diff
@@ -132,5 +134,5 @@ fn load_items_from_file(file_path: &str) -> Result<Items, Box<dyn Error>> {
     let mut file_content = String::new();
     file.read_to_string(&mut file_content)?;
 
-    Ok(toml::from_str(&file_content)?)
+    Ok(toml::from_str(&file_content).expect("Toml format was not valid"))
```
Comment (jnferner) [inline src/item_loader.rs]: You sure about that?

File: src/item_loader.rs [hunk 2]
```diff
@@ -114,16 +114,18 @@ impl ItemLoader for TomlItemLoader {
...
-fn load_items_from_entries(entries: impl Iterator<Item = DirEntry>) -> HashMap<String, Item> {
-    entries
+fn load_items_from_entries(entries: impl Iterator<Item = DirEntry>) -> Result<HashMap<String, Item>, Box<dyn Error>> {
+    let items = entries
         .map(|entry| entry.path().to_string_lossy().into_owned())
         .filter(|file_path| file_path.ends_with(".toml"))
-        .map(|file_path| load_items_from_file(&file_path).into_iter())
-        .flatten()
+        .map(|file_path| load_items_from_file(&file_path))
+        .collect::<Result<Vec<_>, Box<dyn Error>>>()?
```
Comment (jnferner) [inline src/item_loader.rs]: Does type inference work on the Error type here?

File: src/item_loader.rs (hunk 1 above)
Comment (bash) [reply src/item_loader.rs]: Oops, that one shouldn't have been committed 🙈. Reverted in af9696a

File: src/item_loader.rs (hunk 2 above)
Comment (bash) [reply src/item_loader.rs]: 60b2d5e

**Pull request merged**